### PR TITLE
Add Circle CI integration

### LINKED ISSUE #31: Add CI integration

### PR DESCRIPTION
**Pull request by bergundy**
Closes #31

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          name: Install dependencies
+          command: node common/scripts/install-run-rush.js install
+      - save_cache:
+          key: dependency-cache-{{ checksum "common/config/rush/shrinkwrap.yaml" }}
+          paths:
+            - common/temp
+      - run:
+          name: Build source
+          command: node common/scripts/install-run-rush.js rebuild --verbose
+      # - run:
```
Comment (arikmaor) [inline .circleci/config.yml]: Is this commented out on purpose?

File: babel-macro/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   ],
   "description": "",
   "scripts": {
-    "build": "rm -rf dist && tsc && cp -R 'src/test/backend/' 'dist/test/backend'",
+    "build": "rm -rf dist && tsc && cp -r src/test/backend/ dist/test/",
```
Comment (arikmaor) [inline babel-macro/package.json]: From `man cp`
```
COMPATIBILITY
	Historic versions of the cp utility had a -r option.
	This implementation supports that option; however, its use is strongly discouraged, as it does not correctly copy special files, symbolic links, or fifo's.
```

File: babel-macro/src/test/macro.spec.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ process.chdir(__dirname);
 pluginTester({
   plugin,
   babelOptions: {
-    filename: join(__filename, 'tested-file.js'),
+    filename: join(__dirname, 'babel-output', 'tested-file.js'),
```
Comment (arikmaor) [inline babel-macro/src/test/macro.spec.ts]: The tests are passing after this change?

File: .circleci/config.yml (hunk 1 above)
Comment (bergundy) [reply .circleci/config.yml]: Yes, I'll add a task to enable linting on all repos.

File: babel-macro/package.json (hunk 2 above)
Comment (bergundy) [reply babel-macro/package.json]: It's not in the linux man page http://man7.org/linux/man-pages/man1/cp.1.html

File: babel-macro/src/test/macro.spec.ts (hunk 3 above)
Comment (bergundy) [reply babel-macro/src/test/macro.spec.ts]: Yes, you can see the status on the PR.

File: babel-macro/package.json (hunk 2 above)
Comment (arikmaor) [reply babel-macro/package.json]: So you're saying that you *want to stop* supporting development on mac? (as you are specifically changing from the cross-platform option to the linux only option and this is the only actual change that is happening here)

I don't want to block, so I'm approving but from now on I'll take under-consideration that building on our machines is not necessary nor supporting developers who wants to contribute but don't know how to use docker...

Comment (vogre) [reply babel-macro/package.json]: I think the changes between BSD and Linux can be ignored, and something like `cp -a` would be unfamiliar.

Comment (bergundy) [reply babel-macro/package.json]: Just note that this still works on Mac..

Comment (arikmaor) [reply babel-macro/package.json]: Current state: using -R
Proposed change: using -r
difference in linux - none, they both work the same
difference in osx - going from recommended to deprected
difference in windows - none, they both don't work

So this change as I see it, is only to stick it to apple.
And it's an explicit change just for that, it causes no other change.

**Pull request merged**